tiatoolbox/models/architecture/micronet.py

```diff
@@ -309,25 +330,33 @@ def group4_arch_branch(in_ch, out_ch, up_kernel=(2, 2), up_strides=(2, 2)):
             padding=0,
             bias=True,
         ),
-        nn.Tanh(),
+        activation,
     )
     return nn.ModuleDict(module_dict)
 
 
-def out_arch_branch(in_ch, num_class=2):
+def out_arch_branch(in_ch: int, num_class: int = 2, activation: str = "softmax"):
     """Group5 branch for MicroNet.
 
+    This branch defines architecture for auxiliary and the main output.
+
     Args:
         in_ch (int):
             Number of input channels.
         num_class (int):
             Number of output channels. default=2.
+        activation (str):
+            Activation function, default="softmax".
 
     Returns:
         torch.nn.Sequential:
             An output of type :class:`torch.nn.Sequential`
 
     """
+    if activation == "relu":
```

Comment:
it seems that you only accept "relu" and "softmax" activations, so I would suggest you check the input for these two types instead of `if...else...` statement.